src/kakarot/instructions/system_operations.cairo

```diff
@@ -1035,7 +1035,15 @@ namespace SystemOperations {
         tempvar extra_gas = access_gas_cost + create_gas_cost + transfer_gas_cost;
         let evm = EVM.charge_gas(evm, extra_gas + memory_expansion.cost);
 
-        let gas = Gas.compute_message_call_gas(gas_param, evm.gas_left);
+        let (is_gas_zero) = uint256_eq(gas_param, Uint256(0, 0));
+        if (is_gas_zero != FALSE) {
+            let (high, low) = split_felt(evm.gas_left);
+            tempvar gas_limit = Uint256(low, high);
+        } else {
+            tempvar gas_limit = gas_param;
+        }
+        tempvar gas_limit = gas_limit;
```

Comment:
`let gas_limit = [ap - 1]`

Reply:
let gas_limit = Uint256([ap-2], [ap-1]) ?

Reply:
I think that `cast([ap - 2], Uint256)` works as well